mozjs/src/gc/root.rs

```diff
@@ -216,6 +216,11 @@ impl<'a, T> MutableHandle<'a, T> {
         unsafe { *self.ptr = v }
     }
 
+    /// Safety: GC must not run during the lifetime of the returned reference.
+    pub unsafe fn as_mut<'b: 'a>(&mut self) -> &'b mut T {
```

Comment:
I think those lifetimes are wrong. We do not want for `'b` to outlive `'a`, right?

Rust lifetimes `'long: 'short` are read as "'long outlives 'short".`

Reply:
And these are right: https://github.com/servo/mozjs/blob/3e8e9d38531e74c75b3ad4921e895c9e8a496082/mozjs/src/gc/root.rs#L229-L232

Reply:
You're right, this is backwards... I'm also not sure why I thought this was a good idea. The lifetime should be bounded by the lifetime of the mutable reference to self, not just the lifetime of the data it points at. Will fix.

Reply:
Updated again as per discussion in #569